### DISCUSSION TITLE
Can I use MariaDB for MySQL?

### DISCUSSION THREAD

**Discussion by mkastner** [keystonejs/keystone#8145] [Questions] [answered]
Can I use MariaDB as drop-in replacement for MySql?

**Comment by flexdinesh** [comment, marked as answer]
~Not at the moment. Keystone only supports 3 database types - [db config](https://keystonejs.com/docs/config/config#db).~

Yes, our automated tests are actually running against `mariadb:10.10`.

**Comment by mkastner** [comment]
@flexdinesh Thanks for your answer. Sorry, I noticed it quite late. That's great!

**Comment by lmiller1990** [comment]
Has anyone does this? I am using MariaDB, we use binary(16) which is a compressed UUID, for the primary key and keystone seems to not know how to handle the binary data for queries.

I get this error:

```
Error in domain logic: Conversion from `Bytes([0, 12, 2, 183, 253, 164, 71, 8, 168, 183, 95, 179, 67, 119, 11, 54])` to `String` failed.
```

I tried to configure the list like this:

```js
  marker: list({
    db: {
      idField: { // <============= here
        kind: "random",
        bytes: 16,
        encoding: "hex",
      },
    },
    access: allowAll,

    // this is the fields for our User list
    fields: {
      // by adding isRequired, we enforce that every User should have a name
      //   if no name is provided, an error will be displayed
      name: text({ validation: { isRequired: true } }),
    },
  }),
```

But no combination of `idField` seems to work.
